src/lib/ndpi_main.c

```diff
@@ -4136,6 +4136,10 @@ int ndpi_handle_ipv6_extension_headers(struct ndpi_detection_module_struct *ndpi
       }
 
       *nxt_hdr = (*l4ptr)[0];
+      frag_offset = ntohs(*(u_int16_t *)((*l4ptr) + 2)) >> 3;
```

Comment:
Just a question: `l4ptr` does not point to any layer 4 protocol but instead to an IPv6 extension header or am I wrong?

Reply:
Inside the function, `l4ptr` always points to the next header in the chain (a Ipv6 extension header or a proper L4 header).
Returning from the function, it points to the L4 header. I *think* this is the reason this parameter has been named in this way